test/json_unittest.cc

```diff
@@ -0,0 +1,428 @@
+  });
+}
+
+TEST(TrivialParseTest, False) {
+  GuardJsonException([](){
+    json::value v = json::Parser::FromString("false");
+    EXPECT_EQ(false, v->As<json::Boolean>()->value());
+  });
+}
+
+TEST(TrivialToStringTest, False) {
+  GuardJsonException([](){
+    EXPECT_TOSTRING_EQ("false", json::boolean(false));
+  });
+}
+
+
+// Trival number.
+
+TEST(TrivialParseTest, Number) {
+  GuardJsonException([](){
+    json::value v = json::Parser::FromString("2");
+    EXPECT_EQ(2.0, v->As<json::Number>()->value());
+  });
+}
+
+TEST(TrivialToStringTest, Number) {
+  GuardJsonException([](){
+    EXPECT_TOSTRING_EQ("2.0", json::number(2.0));
+  });
+}
+
+
+// Trival string.
+
+TEST(TrivialParseTest, StringEmpty) {
+  GuardJsonException([](){
+    json::value v = json::Parser::FromString("\"\"");
+    EXPECT_EQ("", v->As<json::String>()->value());
+  });
+}
+
+TEST(TrivialToStringTest, StringEmpty) {
+  GuardJsonException([](){
+    EXPECT_TOSTRING_EQ("\"\"", json::string(""));
+  });
+}
+
+TEST(TrivialParseTest, StringOneChar) {
+  GuardJsonException([](){
+    json::value v = json::Parser::FromString("\"x\"");
+    const auto& str = v->As<json::String>();
+    EXPECT_EQ("x", str->value());
+  });
+}
+
+TEST(TrivialToStringTest, StringOneChar) {
+  GuardJsonException([](){
+    EXPECT_TOSTRING_EQ("\"x\"", json::string("x"));
+  });
+}
+
+
+// Trival array.
+
+TEST(TrivialParseTest, ArrayEmpty) {
+  GuardJsonException([](){
+    json::value v = json::Parser::FromString("[]");
+    EXPECT_TRUE(v->As<json::Array>()->empty());
+  });
+}
+
+TEST(TrivialToStringTest, ArrayEmpty) {
+  GuardJsonException([](){
+    EXPECT_TOSTRING_EQ("[]", json::array({}));
+  });
+}
+
+TEST(TrivialParseTest, ArrayOneElement) {
+  GuardJsonException([](){
+    json::value v = json::Parser::FromString("[2]");
+    const auto& arr = v->As<json::Array>();
+    EXPECT_EQ(1, arr->size());
+    EXPECT_EQ(2.0, (*arr)[0]->As<json::Number>()->value());
+  });
+}
+
+TEST(TrivialToStringTest, ArrayOneElement) {
+  GuardJsonException([](){
+    EXPECT_TOSTRING_EQ("[2.0]", json::array({json::number(2.0)}));
+  });
+}
+
+
+// Trival object.
+
+TEST(TrivialParseTest, ObjectEmpty) {
+  GuardJsonException([](){
+    json::value v = json::Parser::FromString("{}");
+    EXPECT_TRUE(v->As<json::Object>()->empty());
+  });
+}
+
+TEST(TrivialToStringTest, ObjectEmpty) {
+  GuardJsonException([](){
+    EXPECT_TOSTRING_EQ("{}", json::object({}));
+  });
+}
+
+TEST(TrivialParseTest, ObjectOneField) {
+  GuardJsonException([](){
+    json::value v = json::Parser::FromString("{\"f\":2}");
+    const auto& obj = v->As<json::Object>();
+    EXPECT_EQ(1, obj->size());
+    EXPECT_TRUE(v->As<json::Object>()->Has("f"));
+    EXPECT_FALSE(v->As<json::Object>()->Has("g"));
+    EXPECT_EQ(2.0, obj->Get<json::Number>("f"));
+  });
+}
+
+TEST(TrivialToStringTest, ObjectOneField) {
+  GuardJsonException([](){
+    json::value v = json::object({
+      {"f", json::number(2.0)}
```

Comment:
Let's add a trailing comma, to indicate our preference to that style.

Reply:
Done.